### PR TITLE
Editor removal

### PR DESCRIPTION
**Pull request by NateCastleOIT**
I've added the ability to delete/mass delete walls with these keybinds:
While in EditingMode:
(Hold) D:
Left click wall = mark wall. Marked walls change colors.
Right click wall = unmark wall. if already unmarked, does nothing.
Enter = deletes all marked walls and adds them to a 'deletedWalls' list

Ctrl + Z = Undo last change. This will remove the most recently added wall to the deleteWalls list.
Ctrl + Y = Redo last change. This will remove the last wall in the stage's walls and add it to the deletedWalls list.

### CODE REVIEW

File: TKGame/Level Editor Content/LevelEditor.cs [hunk 1]
```diff
@@ -99,13 +101,83 @@ public static void BuildWall(Stage stage, GraphicsDevice graphics, SpriteBatch s
...
+        {
+            foreach (var wall in walls)
+            {
+                // Check each wall to see if the mouse is over it
+                if (wall.HitBox.Contains(Input.MouseState.Position))
+                {
+                    // If the Left mouse button was clicked, highlight it with a different color and add it to to be deleted walls
+                    if (Input.MouseState.LeftButton == ButtonState.Pressed && !deletedWalls.Contains(wall))
+                    {
+                        wall.Texture.SetData<Color>(new Color[] { Color.OrangeRed });
```
Comment (keeganhauser) [inline TKGame/Level Editor Content/LevelEditor.cs:120]: might be worth it to place this color as a variable

File: TKGame/Level Editor Content/LevelEditor.cs [hunk 2]
```diff
@@ -99,13 +101,83 @@ public static void BuildWall(Stage stage, GraphicsDevice graphics, SpriteBatch s
...
+                        deletedWalls.Add(wall);
+                    }
+                    // If Right clicked, return the color to White and remove it from to be deleted walls
+                    else if (Input.MouseState.RightButton == ButtonState.Pressed && deletedWalls.Contains(wall))
+                    {
+                        wall.Texture.SetData<Color>(new Color[] { Color.White });
```
Comment (keeganhauser) [inline TKGame/Level Editor Content/LevelEditor.cs:126]: ditto with the color as variable

File: TKGame/Player.cs [hunk 3]
```diff
@@ -1,32 +1,36 @@
...
+
+        internal bool isJumping = false;
+
         public static Player Instance
         {
             get
             {
                 // Creates the player if it doesn't already exist
-                // Uses thread locking to guarentee safety.
+                // Uses thread locking to guarantee safety.
```
Comment (keeganhauser) [inline TKGame/Player.cs:26]: pog typo fix

**Review by keeganhauser — APPROVED** [review]
looks good my g

**Review by NathanGr33n — APPROVED** [review]
Looks good to me, a lot of these changes seem similar between the two pull requests. Don't know if that will cause problems when merging

**Pull request merged**